### PR TITLE
update some nuget packages and bump oryx to dotnet core 2.2 

### PR DESCRIPTION
**Pull request by arroyc**
the reason is to get rid of security alerts in daily builds

### CODE REVIEW

File: images/build/Dockerfile [hunk 1]
```diff
@@ -133,7 +134,7 @@ RUN set -ex \
  && ln -s 3.0 3 \
  && ln -s $sdksDir/$DOT_NET_CORE_30_SDK_VERSION $NET_CORE_APP_30/sdk \
  # LTS sdk <-- LTS runtime's sdk
- && ln -s 2.1 lts \
+ && ln -s 2.2 lts \
```
Comment (gpcastro) [inline images/build/Dockerfile]: 2.1 is the actually LTS - https://dotnet.microsoft.com/download/dotnet-core

File: src/BuildScriptGenerator/BuildScriptGenerator.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (gpcastro) [inline src/BuildScriptGenerator/BuildScriptGenerator.csproj]: I'm confused why this was the fix, since we're updating it to 2.2, but 2.1 is .NET Core LTS. Is it possible that the issue is the 2.1 patch version in the build machines that is the issue, and not the fact that it is 2.1?

File: tests/Oryx.Common.Tests/StringExtensionsTest.cs [hunk 3]
```diff
@@ -63,7 +63,7 @@ public void ReplaceUrlUserInfo_Sanity()
 
             Assert.Equal(
                 "before https://***@example.com/ and then ftp://***@example.org/subdir",
-                "before https://bla:123bla@example.com/ and " +
+                "before https://bla:123bla@example.com/ and ".ReplaceUrlUserInfo() +
```
Comment (dorfire) [inline tests/Oryx.Common.Tests/StringExtensionsTest.cs]: did this test fail before this change?

File: images/build/Dockerfile (hunk 1 above)
Comment (arroyc) [reply images/build/Dockerfile]: thanks .. I am reverting this change

File: src/BuildScriptGenerator/BuildScriptGenerator.csproj [hunk 4]
```diff
@@ -27,7 +27,7 @@
     <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
     <PackageReference Include="Scriban.Signed" Version="1.2.9" />
     <PackageReference Include="SemanticVersioning" Version="1.2.0" />
-    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
+    <PackageReference Include="StyleCop.Analyzers" Version="1.0.2">
```
Comment (dorfire) [inline src/BuildScriptGenerator/BuildScriptGenerator.csproj]: what's this change for?

File: tests/Oryx.Common.Tests/StringExtensionsTest.cs [hunk 5]
```diff
@@ -63,7 +63,7 @@ public void ReplaceUrlUserInfo_Sanity()
 
             Assert.Equal(
                 "before https://***@example.com/ and then ftp://***@example.org/subdir",
-                "before https://bla:123bla@example.com/ and ".ReplaceUrlUserInfo() +
+                "before https://bla:123bla@example.com/ and " +
```
Comment (dorfire) [inline tests/Oryx.Common.Tests/StringExtensionsTest.cs]: did the test fail before this change?

File: tests/Oryx.Common.Tests/StringExtensionsTest.cs (hunk 3 above)
Comment (arroyc) [reply tests/Oryx.Common.Tests/StringExtensionsTest.cs]: reverted this change

File: src/BuildScriptGenerator/BuildScriptGenerator.csproj (hunk 4 above)
Comment (arroyc) [reply src/BuildScriptGenerator/BuildScriptGenerator.csproj]: reverted

File: tests/BuildScriptGenerator.Tests/BuildScriptGenerator.Tests.csproj [hunk 6]
```diff
@@ -12,7 +12,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.1.1" />
     <PackageReference Include="Moq" Version="4.10.1" />
     <PackageReference Include="xunit" Version="2.3.1" />
```
Comment (kichalla) [inline tests/BuildScriptGenerator.Tests/BuildScriptGenerator.Tests.csproj]: update all xunit versions to 2.4.1

File: images/build/Dockerfile [hunk 7]
```diff
@@ -95,7 +95,8 @@ RUN set -ex \
  && cd $sdksDir \
  && ln -s 1.1 1 \
  && ln -s 2.1 2 \
- && ln -s 3.0 3
+ && ln -s 2.2 3 \
+ && ln -s 3.0 4
```
Comment (kichalla) [inline images/build/Dockerfile]: revert these changes
